contracts/TransmuteDPOS.sol

```diff
@@ -92,5 +117,42 @@ contract TransmuteDPOS is TransmuteToken, RoundManager, ProviderPool {
     if (p.status == ProviderStatus.Registered) {
       updateProvider(_provider, p.totalAmountBonded);
     }
+    emit DelegatorBonded(msg.sender, _provider, _amount);
+  }
+
+  function unbond() external {
+    // Only Bonded Delegators can call the function
+    require(delegatorStatus(msg.sender) == DelegatorStatus.Bonded);
+    // TODO: What if a Provider calls unbond() on himself ?
+    // Should he resign ?
+    // What about the tokens of the Delegators that bonded to him ?
```

Comment:
Refunded - but who pays the gas cost for that? Him, I suppose.

Reply:
Yeah, but that could potentially be a huge cost in gas.
Otherwise we could just let them unbond() by themselves if they received the event that their Provider resigned.

Reply:
How does Livepeer handle this?

Reply:
They resign the Transcoder without refunding the delegators.